src/tools/kdb/CMakeLists.txt

```diff
@@ -7,27 +7,33 @@ add_toolheaders (HDR_FILES)
 include_directories (${CMAKE_CURRENT_SOURCE_DIR})
 
 file (GLOB SRC_FILES *.cpp)
-
 set (SOURCES ${SRC_FILES} ${HDR_FILES})
 
+if (BUILD_SHARED OR BUILD_FULL)
+    add_library (kdb-objects OBJECT ${SRC_FILES})
+endif (BUILD_SHARED OR BUILD_FULL)
+
 if (BUILD_SHARED)
-	add_executable (kdb ${SOURCES})
+    add_executable (kdb $<TARGET_OBJECTS:kdb-objects>)
+    add_dependencies (kdb kdberrors_generated)
 
-	target_link_libraries (kdb elektra-core elektra-kdb elektratools)
+    target_link_libraries (kdb elektra-core elektra-kdb elektratools)
 
-	install (TARGETS kdb DESTINATION bin)
+    install (TARGETS kdb DESTINATION bin)
 endif (BUILD_SHARED)
 
 if (BUILD_FULL)
-	add_executable (kdb-full ${SOURCES})
+    add_executable (kdb-full $<TARGET_OBJECTS:kdb-objects>)
+    add_dependencies (kdb-full kdberrors_generated)
 
-	target_link_libraries (kdb-full elektra-full elektratools-full)
+    target_link_libraries (kdb-full elektra-full elektratools-full)
 
-	install (TARGETS kdb-full DESTINATION bin)
+    install (TARGETS kdb-full DESTINATION bin)
 endif (BUILD_FULL)
 
 if (BUILD_STATIC)
 	add_executable (kdb-static ${SOURCES})
+	add_dependencies (kdb-static kdberrors_generated)
```

Comment:
I can't explain why some of those lines are marked as new, because in the [current version](https://github.com/ElektraInitiative/libelektra/blob/master/src/tools/kdb/CMakeLists.txt) they are there already. Any idea?